scheduler/util.go

```diff
@@ -49,50 +100,53 @@ func ScheduleSingleTaskWorker(prefix string, c config.Worker, t *tes.Task) *Offe
 	tzones := t.GetResources().GetZones()
 	project := t.GetProject()
 
-	w := &pbf.Worker{
+	n := &pbs.Node{
 		Id: prefix + t.Id,
-		Resources: &pbf.Resources{
+		Resources: &pbs.Resources{
 			Cpus:   cpus,
 			RamGb:  ram,
 			DiskGb: disk,
 		},
 	}
 
-	w.Metadata = map[string]string{"project": project}
+	n.Metadata = map[string]string{"project": project}
 
 	if len(tzones) >= 1 {
 		// TODO figure out zone mapping when len(tzones) > 1
-		w.Zone = tzones[0]
+		n.Zone = tzones[0]
 	}
 
-	return NewOffer(w, t, Scores{})
+	return NewOffer(n, t, Scores{})
 }
 
-// SetupTemplatedHPCWorker sets up a worker in a HPC environment with a shared
+// SetupTemplatedHPCNode sets up a node in a HPC environment with a shared
 // file system. It generates a submission file based on a template for
 // schedulers such as SLURM, HTCondor, SGE, PBS/Torque, etc
-func SetupTemplatedHPCWorker(name string, tpl string, conf config.Config, w *pbf.Worker) (string, error) {
+func SetupTemplatedHPCNode(name string, tpl string, conf config.Config, n *pbs.Node) (string, error) {
 	var err error
 
+	nconf := conf.Scheduler.Node
+
 	// TODO document that these working dirs need manual cleanup
-	workdir := path.Join(conf.Worker.WorkDir, w.Id)
+	workdir := path.Join(nconf.WorkDir, n.Id)
 	workdir, _ = filepath.Abs(workdir)
 	err = util.EnsureDir(workdir)
 	if err != nil {
 		return "", err
 	}
 
-	wc := conf
-	wc.Worker.ID = w.Id
-	wc.Worker.Timeout = 5 * time.Second
-	wc.Worker.Resources.Cpus = w.Resources.Cpus
-	wc.Worker.Resources.RamGb = w.Resources.RamGb
-	wc.Worker.Resources.DiskGb = w.Resources.DiskGb
+	nconf.ID = n.Id
+	nconf.Timeout = 5 * time.Second
+	nconf.Resources.Cpus = n.Resources.Cpus
+	nconf.Resources.RamGb = n.Resources.RamGb
+	nconf.Resources.DiskGb = n.Resources.DiskGb
 
-	confPath := path.Join(workdir, "worker.conf.yml")
-	wc.ToYamlFile(confPath)
+	c := conf
+	c.Scheduler.Node = nconf
```

Comment:
If you want, you could make nconf a pointer, remove the copy of conf (to `c`) and write `conf` to the file.

Reply:
Also, why does the node now need the full stack config?